effects/ox/src/main/scala/sttp/client4/ox/ws/OxWebSockets.scala

```diff
@@ -0,0 +1,53 @@
+package sttp.client4.ox.ws
+
+import ox.*
+import ox.channels.Channel
+import ox.channels.ChannelClosed
+import ox.channels.Sink
+import ox.channels.Source
+import sttp.client4.ws.SyncWebSocket
+import sttp.ws.WebSocketFrame
+
+import scala.util.control.NonFatal
+
+  extension (ws: SyncWebSocket)(using Ox) def asSourceAndSink: (Source[WebSocketFrame], Sink[WebSocketFrame]) = 
```

Comment:
maybe simply a function, I'm not sure if the extension method here gives us anything? you have to deconstruct it anyway.